src/sqlitestorage.cpp

```diff
@@ -198,63 +198,69 @@ bool SqliteStorage::open()
   sqlite3_busy_timeout( d->mDatabase, 1500 );
 
   /* Create Calendars, Components, etc. tables */
-  query = CREATE_VERSION;
-  sqlite3_exec( d->mDatabase );
+#ifdef Q_OS_UNIX
+  if (d->mSem.isInitialProcess()) {
```

Comment:
You should be doing this in a mutexed state - creating the DB is a form of writing to it.  If you already have the mutex (which it looks like to me), then nothing else can enter this code until the DB has been created by the initial process and the lock subsequently released.